src/tests/sync/src/test_utils.rs

```diff
@@ -0,0 +1,155 @@
+// Copyright 2022 Singularity Data
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::process::{Command, Output};
+use std::sync::Arc;
+use std::thread::JoinHandle;
+use std::time::Duration;
+
+use risingwave_cmd_all::playground;
+use risingwave_common::util::sync;
+use risingwave_common::util::sync::{Signal, WaitForSignal};
+use risingwave_object_store::object::object_metrics::ObjectStoreMetrics;
+use risingwave_object_store::object::{parse_remote_object_store, ObjectStoreImpl};
+use risingwave_pb::common::WorkerType;
+use risingwave_rpc_client::MetaClient;
+use serial_test::serial;
+
+pub fn setup_env() {
+    let current_dir =
+        std::env::var("RW_WORKSPACE").expect("set env RW_WORKSPACE to project root path");
+    std::env::set_current_dir(current_dir).unwrap();
+    std::env::set_var("RW_META_ADDR", "http://127.0.0.1:5690");
+    std::env::set_var("FORCE_SHARED_HUMMOCK_IN_MEM", "1");
+    std::env::set_var("PLAYGROUND_PROFILE", "playground-test");
+    std::env::set_var("OBJECT_STORE_URL", "memory-shared");
+    std::env::set_var("OBJECT_STORE_BUCKET", "hummock_001");
+}
+
+pub async fn start_cluster() -> (JoinHandle<()>, std::sync::mpsc::Sender<()>) {
+    wipe_object_store().await;
+    sync::activate_sync_point(
+        "CLUSTER_READY",
+        vec![sync::Action::EmitSignal("SIG_CLUSTER_READY".to_owned())],
+        1,
+    );
+
+    let (tx, rx) = std::sync::mpsc::channel();
+    let join_handle = std::thread::spawn(move || {
+        let runtime = tokio::runtime::Builder::new_multi_thread()
+            .worker_threads(1)
+            .enable_all()
+            .build()
+            .unwrap();
+        runtime.block_on(async {
+            tokio::spawn(async { playground().await });
+            rx.recv().unwrap();
+        });
```

Comment:
To be precise on the behaviour here:
1. Current thread is blocked on block_on. Current thread's task is blocked on rx.recv().
2. tokio::spawn must spawn in worker thread.
3. When rx.recv() is unblocked, block_on completes. Thus, the thread is unblocked, thus dropping the tokio runtime, thus, dropping the worker thread and the spawned tasks.

Is this correct?

Reply:
Yes. This is a workaround to shutdown playground, cause playground doesn't provide one yet.